cpp/comms/mpi/CMakeLists.txt

```diff
@@ -18,17 +18,25 @@ cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 project(cuML-comms-MPI LANGUAGES CXX CUDA)
 
 find_package(MPI REQUIRED)
-find_package(NCCL)
+
+if(NOT NCCL_PATH)
+  find_package(NCCL REQUIRED)
+else()
+  set(NCCL_INCLUDE_DIRS ${NCCL_PATH}/include)
```

Comment:
From my understanding the else path should be handled by FindNCCL. If it is not already the case FindNCCL should be used to pick up `NCCL_PATH` to find NCCL.

Reply:
The find(nccl) extensión was not working properly in certain scenarios with conda environments when nccl was also present in the system. So this approach works reliably until/unless we want to modify/rewrite the findnccl.cmake which probably should be a future PR if that’s the case

Reply:
If I recall correctly that happened in certain scenarios where find nccl would prioritize a system installed 1.x nccl (in spite of using hints, which was the main issue), so this approach allows users (and CI) to fully control the path and avoid that issue with hints

Reply:
I must admit that I just shamelessly copied this from the corresponding code in `std/CMakeLists.txt`. Didn't realize that it was more of a reliability issue. Have filed issue #1263 for this.

Reply:
Thanks! You ninja’d me to writing the issue! Will repro the issue and post the exact scenario where that happens in there. Want to merge this fix quickly though, that ok?

Reply:
I'm ok, provided @jirikraus is ok with addressing this in a separate PR fixing issue #1263 .

Reply:
LGTM